massa-proto/proto/massa/api/v1/execution.proto

```diff
@@ -67,16 +70,226 @@ message ScExecutionEventContext {
 
 // ScExecutionEventStatus type enum
 enum ScExecutionEventStatus {
-  SC_EXECUTION_EVENT_STATUS_UNSPECIFIED = 0; // Defaut enum value
+  SC_EXECUTION_EVENT_STATUS_UNSPECIFIED = 0; // Default enum value
   SC_EXECUTION_EVENT_STATUS_FINAL = 1; // Final status
   SC_EXECUTION_EVENT_STATUS_READ_ONLY = 2; // Read only status
   SC_EXECUTION_EVENT_STATUS_FAILURE = 3; // Failure status
 }
 
 // ExecutionOutputStatus type enum
 enum ExecutionOutputStatus {
-  EXECUTION_OUTPUT_STATUS_UNSPECIFIED = 0; // Defaut enum value
+  EXECUTION_OUTPUT_STATUS_UNSPECIFIED = 0; // Default enum value
   EXECUTION_OUTPUT_STATUS_CANDIDATE = 1; // Candidate status
   EXECUTION_OUTPUT_STATUS_FINAL = 2; // Final status
 }
- 
\ No newline at end of file
+
+// StateChanges
+message StateChanges {
+  // Ledger changes
+  LedgerChanges ledger_changes = 1;
+  // Asynchronous pool changes
+  repeated AsyncPoolChangeEntry async_pool_changes = 2;
+  // Executed operations changes
+  ExecutedOpsChanges executed_ops_changes = 5;
+}
+
+// ExecutedOpsChanges
+message ExecutedOpsChanges {
+  // Executed operations
+  repeated ExecutedOpsChangeEntry executed_ops = 1;
+}
+
+// ExecutedOpsChangeEntry
+message ExecutedOpsChangeEntry {
+  // OperationId
+  string operation_id = 1;
+  // ExecutedOpsChangeValue
+  ExecutedOpsChangeValue value = 2;
+}
+
+// ExecutedOpsChangeValue
+message ExecutedOpsChangeValue {
+  // The status of the execution of the operation
+  repeated OperationExecutionStatus status = 1;
```

Comment:
In my mind we only have one status but maybe Im saying shit. Invoking @bilboquet

Reply:
This is a choice made in gRPC API to avoid breaking changes in the future. We avoid using booleans to handle status.

Reply:
I think that Aurelien was asking about the presence of  the `repeated` keyword.
Even if this subject is still not that clear to me (I think that an operation can be executed multiple times) I think that, else you want to represent the history of execution, there is only one status at a given point in time.
Don't take this for granted, only my understanding, and I'd be glad to read your opinion

Reply:
Oki, thanks, I think I got it, so it looks good to me.

Reply:
Using an array is a cheap way to get the flexibility and avoid breaking changes. In production, any detail counts. 
```
data.status == "FINAL" (enum approch)
data.status.contains("FINAL")
data.status.containsOneOf("FAILED", "SCHEDULED")
```

Reply:
You could do what you went internally and keep the gRPC API non breakable if we adopt this way

Reply:
Let's continue this discussion in an issue to not block the PR.